data/moves.ts

```diff
@@ -16242,10 +16242,16 @@ export const Moves: {[moveid: string]: MoveData} = {
 		secondary: {
 			dustproof: true,
 			chance: 100,
-			onHit(target) {
-				if (target.status === 'brn') target.cureStatus();
+			onHit(target, source, move) {
+				if (target.status === 'brn') {
+					target.addVolatile('cureburn', source, move);
```

Comment:
Why are you using a volatile to track this? Does Sparkling Aria not cure burns if the target gets burned during the attack?

Reply:
the target does not get cured if the user of the move faints on the same turn. I needed the volatile so it still is affected by sheer force but not shield dust, but the actual curing is after "onDamage" as in innards out

Reply:
Oh, it's for Sheer Force.